### PR TITLE
Add and run tflint on tf examples

### PR DESCRIPTION
**Pull request by NissesSenap**
* Add.tflint.hcl
* Run lint on
  * github
  * install
  * multi-env
* Update .gitignore to ignore .terraform.lock.hcl
* run terraform fmt
* Add lint to make
* Add make lint to github action e2e

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -32,3 +35,21 @@ docs: tools
...
 tools:
 	GO111MODULE=on go install github.com/hashicorp/terraform-plugin-docs/cmd/tfplugindocs
+
+.SILENT:
+lint:
+	set -e
+	echo lint: Start
+	MODULE_GROUPS=$$(find examples -mindepth 1 -maxdepth 1 -type d)
+	delete=examples/data-sources
+	echo ${array[@]/$delete}
```
Comment (phillebaba) [inline Makefile:45]: I get that we need to remove the data sources as they will never pass the lint check. Can you capitalize the variable names?

**Comment by phillebaba** [comment]
@NissesSenap see my comment, could you also update the branch?

File: Makefile (hunk 1 above)
Comment (NissesSenap) [reply Makefile]: Done, also changed the name on the variables and made the the actual script working :)

**Comment by NissesSenap** [comment]
@phillebaba fixed

**Comment by phillebaba** [comment]
I am going to hold off with merging this as there are two other PRs that will be blocked if this is merged before. As soon as they are merged I will merge this.

**Review by phillebaba — APPROVED** [review]
LGTM :rose:

**Pull request merged**